terragrunt/aws/network/security-groups.tf

```
@@ -0,0 +1,61 @@
+# 
+# Security Groups for the VPC URL shortener
+#
+
+resource "aws_security_group" "api" {
+
+  name        = "${var.product_name}_api_sg"
+  description = "SG for the URL shortener API"
+
+  vpc_id = module.url_shortener_vpc.vpc_id
+
+  tags = {
+    CostCentre = var.billing_code
+    Name       = "${var.product_name}_api_sg"
+  }
+
+  egress {
+    description = "Allow API outbound connections to the internet"
+    from_port   = 443
+    to_port     = 443
+    protocol    = "TCP"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_security_group" "vpc_endpoint" {
+  name        = "vpc_endpoints"
+  description = "PrivateLink VPC endpoints"
+  vpc_id      = module.url_shortener_vpc.vpc_id
+
+  tags = {
+    Name       = "${var.product_name}_api_sg"
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+}
+
```

We'll need two more `aws_security_group_rule` resources to handle the `logs` and `dynamodb` ingress traffic.  They'll look like so:
```suggestion

resource "aws_security_group_rule" "vpc_endpoint_interface_ingress" {
  description              = "Ingress from the API security group to the private interface endpoints"
  type                     = "ingress"
  from_port                = 443
  to_port                  = 443
  protocol                 = "tcp"
  security_group_id        = aws_security_group.vpc_endpoint.id
  source_security_group_id = aws_security_group.api.id
}

resource "aws_security_group_rule" "vpc_endpoint_dynamodb_ingress" {
  description       = "Ingress from the private DynamoDB endpoint"
  type              = "ingress"
  from_port         = 443
  to_port           = 443
  protocol          = "tcp"
  security_group_id = aws_security_group.vpc_endpoint.id
  prefix_list_ids = [
    aws_vpc_endpoint.dynamodb.prefix_list_id
  ]
}

```